fpga_interchange/site_router.cc

```diff
@@ -1315,6 +1315,7 @@ bool SiteRouter::checkSiteRouting(const Context *ctx, const TileStatus &tile_sta
 
     block_lut_outputs(&site_arch, blocked_wires);
     block_cluster_wires(&site_arch);
+    reserve_site_ports(&site_arch);
```

Comment:
we had previously discussed not putting this in the `checkSiteRouting` path for performance reasons. we might need to benchmark and see if it slows things down significantly, if it does then some optimisation (caching etc) will be needed.

Reply:
Right, this was actually a leftover. The issue was that, given that the dedicated placement check was not running for clusters, the IDELAYs in the test-case were placed in invalid locations, therefore triggering an issue. The addition of this to the site routing check was a temporary check.

Reply:
Sure, thanks for explaining!